packages/workbox-core/src/types.ts

```diff
@@ -72,12 +72,23 @@ export interface RouteHandlerObject {
  */
 export type RouteHandler = RouteHandlerCallback | RouteHandlerObject;
 
+export interface HandlerWillStartCallbackParam {
+  request: Request;
+  event?: ExtendableEvent;
+  state?: {[prop: string]: any};
+}
+
+export interface HandlerWillStartCallback {
+  (param: HandlerWillStartCallbackParam): Promise<void | null | undefined>;
```

Comment:
What's the significance to `Promise<void | null | undefined>` here?

I know the difference between `null` and `undefined`, but not sure how `void` comes into play. I'm also not sure why all three are possible values that the we want to support the `Promise` fulfilling with.

Reply:
It's tricky and I'm honestly not sure how best to solve this (I think it's a TypeScript bug/quirk).

`void` is recommended to only be used with functions (to denote no return value), but in the case of async functions that doesn't return a value, you have to use `Promise<void>`. However, as soon as you have any other code that stores the return value of an async function like this, you need to account for some functions explicitly returning undefined and some functions not having a return statement.

In the case of third-party plugin code, I think we need to support all three...

(Let me know if you have any thoughts on how to improve this.)

Reply:
In that case, I wonder whether letting folks explicitly return `undefined` or `null` is necessary vs. just requiring no `return` statement by using `Promise<void>`.

In other words, why do we want folks to be able to explicitly return `undefined` or `null` from their `async` callback function?